CppCoreGuidelines.md

```diff
@@ -13791,168 +13793,174 @@ Performance rule summary:
 
-    vector<uint8_t> v(100000);
+Whatever your system may be, referring back to your needs and your customer
+needs will tell you where you should be investing and which investments are
+worthwhile. At some point, the potential benefits are not worth the opportunity
+cost of doing them.  An extra 10% reduction of download size maybe isn't worth
+the engineering time because you've reached a size that is small enough that
+your customers no longer notice the difference.  But if your needs also include
+egress costs for those downloads then maybe that same 10% is worth it.  The very
+same potential improvement might be a great a idea or a silly one depending on
+your needs.
 
-    for (auto& c : v)
-        c = ~c;
+### <a name="Rper-simple"></a>Per.4: Simpler is generally better
 
-##### Example, bad
+Assuming you have done the basics of making sure that your overall design is approximately right,
+such as you are using O(N) if it is available rather than O(N^2) or something like that then you'll
+find that *simplifying* code and data is the best way to make things faster.  This means:
 
-    // intended to be faster, but is often slower
+* don't create complex inheritance graphs
+* don't use complicate dispatch logic if simple logic will do
+* don't make data structures with lots of pointers
+* don't use shared pointers if unique pointers will do
+* don't use patterns that reduce your average code and data density
 
-    vector<uint8_t> v(100000);
+In short, consider the problem you need to solve and solve it as directly as you reasonably can.
+Don't completely focus on just your immediate needs, some future-proofing is usually necessary, but
+don't ruin the code by over-generalizing it either.  Remember that in order to be *re*-usable code
+has to first be *usable*.
 
-    for (size_t i = 0; i < v.size(); i += sizeof(uint64_t)) {
-        uint64_t& quad_word = *reinterpret_cast<uint64_t*>(&v[i]);
-        quad_word = ~quad_word;
-    }
+This is a bit contentious but in practice, object oriented programming generally
+results in the worst performance. Consider work-oriented programming instead.
+What has to happen, how do we do this clearly, simply, directly, with only as
+much abstraction as would be helpful.
 
-##### Note
+Arrays/vectors are probably the most under-utilized and most important data structure.  Use
+simple proven data structures a lot.
 
-???
-
-???
 
-### <a name="Rper-low"></a>Per.5: Don't assume that low-level code is necessarily faster than high-level code
+### <a name="Rper-measure"></a>Per.5: Don't make claims about performance without measurements
 
 ##### Reason
 
-Low-level code sometimes inhibits optimizations. Optimizers sometimes do marvels with high-level code.
-
-##### Note
-
-???
-
-???
-
-### <a name="Rper-measure"></a>Per.6: Don't make claims about performance without measurements
+The field of performance is littered with myth and bogus folklore.  My favorite aphorism,
```

Comment:
Best to avoid writing in the first person in a document with many authors.

Reply:
I'm too used to blogging.  Thank you.

Reply:
replaced with something 3rd person.  Though I'm quoting myself now... Maybe it's not even worth the aphorism at all.  I would not be too sad if you asked me to delete it entirely.